src/TestingDevice.java

```diff
@@ -1,7 +1,24 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public class Device {
+/**
+ * An implementation of the <a href="http://jodypaul.com/cs/sweprin/deviceProj/api/Device.html">Device</a> class used for testing. Allows for linear, polynomial, and pseudo-random rotations per Spin call.
+ * <p>
+ *     Private constructors allow creation of Devices that always spin a number of times in a linear, polynomial, or pseudo-random relationship.
+ *     <br>
+ *     This is helpful for testing by artificially increasing the chances of disclosed bits to 'follow' a linear or polynomial algorithm, potentially hiding one of them, and making it more difficult to unlock.
+ *     <br>
+ *     <br>
+ *     For example, if we always try to poke the first two bits of a device after a spin, and the spin always rotates 4 times in a device holding 4 circular bits, it will always ask for the same 2 bits.
+ *     <br>
+ *     In the same way, if we ask for every 2 bits in a linear relationship, the Device could be rotating in such a way with each spin that it follows our requested peek sequence, potentially preventing a successful unlocking of the device.
+ * </p>
+ * @author Daniel Dews
+ * @version 1.5
+ * @see <a href="http://jodypaul.com/cs/sweprin/deviceProj/api/Device.html">Device</a>
+ * @see <a href="http://jodypaul.com/cs/sweprin/deviceProj/projectDescription.html">Project Description</a>
+ */
+public class TestingDevice extends Device {
     private Link<Character> head;
```

Comment:
I renamed this class so it won't conflict with testing for the implementation Device jar that was given to us.
It now extends the aforementioned class for easier testing.